### PR TITLE
Run private network with docker tutorial

### PR DESCRIPTION
**Pull request by maria-robobug**
Adds a tutorial page in the **Infrastructure** section on how to run a private XRPL network with Docker.

Please let me know if you think this is a useful page to include for developers, whether the steps are clear or not, and if you have suggestions on what can be added to it.

### CODE REVIEW

**Comment by mDuo13** [comment]
From the link checker:

> Link check report. 532241 links checked.
> 2 broken links found:
> File: out/private-network-with-docker.html Link: /ja/private-network-with-docker.html
> File: out/private-network-with-docker.html Link: ../../concepts/servers/peer-protocol.md

The first one is because the file needs to be in the Japanese target. The second one is an md link and should link the html page (`peer-protocol.html` with no path) instead.

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+  - Core Server
+---
+
+# Run a Private Network with Docker
+
+This tutorial describes how to run a private XRP Ledger network on your computer with [Docker](https://docs.docker.com/get-docker/) and the latest version of [rippled](https://hub.docker.com/r/xrpllabsofficial/xrpld).
+
+While you can easily use the public XRP Testnet servers, running a private network can be useful when trying to understand how the XRP Ledger works, or when testing new features in isolation.
+
+**Caution:** This tutorial is mainly suited for development or testing purposes, and does not involve using real money. You should **not** use this configuration for a production network.
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
**Caution:** This tutorial is suited for development or testing purposes only, and does not involve using real money. You should **not** use this configuration for a production network.
```

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+In this tutorial, you will learn:
+
+- How to set up and configure a _small_ network with three `rippled` validator nodes, including how to generate the keys for each node.
+
+- How to run the network with [Docker Compose](https://docs.docker.com/compose/).
+
+- How to verify the network is up and running.
+
+The diagram below shows a high-level overview of the containerized private network that you will set up.
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
The following diagram shows a high-level overview of the containerized private network that you will set up.
```

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+{{ include_svg("img/xrp-ledger-private-network-docker.svg", "Figure 1: Diagram of a three node containerized private ledger network") }}
+
+## Prerequisites
+
+To follow along with this tutorial, ensure that you have the latest version of **Docker** installed on your preferred platform.
+
+## Generate the Validator Keys
+
+Generate the keys for **each** of your validator nodes by using the `validator-keys` tool provided with `rippled`. The generated keys should be saved in a text file on your computer for later use.
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md:38]: Add a link to the tool ?

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+        cd /opt/ripple/bin &&
+            ./validator-keys create_keys --keyfile /PATH/TO/YOUR/validator-<NUMBER>-keys.json
+
+    Sample output:
+
+        Validator keys stored in /PATH/TO/YOUR/validator-<NUMBER>-keys.json
+
+        This file should be stored securely and not shared.
+
+    **Warning:** In a production or test environment you should follow best practices and store the generated keys in a secure, offline, and recoverable location, such as an encrypted USB flash drive. However, as this is an example for a local development set up, storing the keys on your computer is sufficient.
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
    **Warning:** In a production or test environment you should follow best practices always and store the generated keys in a secure, offline, and recoverable location, such as an encrypted USB flash drive. However, as this tutorial is an example of a local development setup, storing the keys on your computer is sufficient.
```

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+        NmluOEhBU1FLUHVnQkQ2N2tNYVJGR3ZtcEFUSGxHS0pkdkRGbFdQWXk1QXFEZWRGdj
+        VUSmEydzBpMjFlcTNNWXl3TFZKWm5GT3I3QzBrdzJBaVR6U0NqSXpkaXRROD0ifQ==
+
+5. Repeat steps **2-4** for the remaining validator nodes. Once you have generated the keys and tokens for _all_ validators, enter `exit` in your terminal to exit the Docker container.
+
+## Configure the Network
+
+This section describes how to configure the validator nodes in your network.
+
+**Note:** The provided configuration enables the network to retain _some_ ledger history, but the amount of transaction history stored will depend on how long the network has been online.
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
**Note:** The configuration in this tutorial enables the network to retain _some_ ledger history, but the amount of transaction history stored will depend on how long the network has been online.
```

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+
+    xrpl-private-network/
+        ├── validator_1/
+        │   └── config
+        ├── validator_2/
+        │   └── config
+        └── validator_3/
+            └── config
+
+In your terminal, run the command below to create the directories:
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
In your terminal, run the following command to create the directories:
```

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+               "ledger_hash" : "6F9F54903CC4546F7A426CD78AFD68D907F5DC40B1780DF31A662CF65920E49C",
+               "ledger_index" : 51,
+               "status" : "success",
+               "validated" : true
+           }
+        }
+        
+    All validator nodes should respond with the same balance of 1000000000 XRP for the `r9wRwVgL2vWVnKhTPdtxva5vdH7FNw1zPs` account.
+
+## (Optional) Stop the Network
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
## Stop the Network
```

**Review by amarantha-k — COMMENTED** [review]
Added a few minor suggestions.

**Comment by WietseWind** [comment]
I quickly skimmed the contents, focussing on the docker/rippled interaction, looking good 

Related, there may be some inspiration there when it comes to script to fire things up, etc.: did you see this before? https://github.com/UNIC-IFF/ripple-docker-testnet 

It has nice extras like traffic generators, samples, helper scripts, etc. etc. etc.
- by University of Nicosia under UBRI

Always good to have a tutorial so people can learn, but if it's about the actual tooling to get things done, the repo I just mentioned trumps a tutorial for sure. So I'd say there's a good place for both, and referring in the tutorial to the repo makes a lot of sense imo

File: content/infrastructure/rippled/run-private-network-with-docker.md (hunk 3 above)
Comment (maria-robobug) [reply content/infrastructure/rippled/run-private-network-with-docker.md:38]: I didn't want to point users to the tool to further confuse them, as this tutorial is using the `validator-keys` tool included in `rippled`, via Docker (which works no problem). But, when building the tool separately (https://github.com/ripple/validator-keys-tool), I've come across errors. So I'm wondering if the standalone version is not being updated?

A few others have seen errors too: https://github.com/ripple/validator-keys-tool/issues/41

This response makes me less inclined to point users to it as well:
https://github.com/ripple/validator-keys-tool/issues/41#issuecomment-1599820254

Comment (amarantha-k) [reply content/infrastructure/rippled/run-private-network-with-docker.md:38]: Got it. Agree with you, let's not add a link.

File: content/infrastructure/rippled/run-private-network-with-docker.md [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+         ✔ Container validator_2                 Removed                                                       1.6s
+         ✔ Network xrpl-private-network_default  Removed                                                       0.0s
+
+## See Also
+
+- **Networks and Servers:**
+    - [Peer Protocol](peer-protocol.html)
+    
+- **References:**
+    - [XRPL Testnet Setup Scripts for Docker](https://github.com/UNIC-IFF/ripple-docker-testnet)
```
Comment (amarantha-k) [inline content/infrastructure/rippled/run-private-network-with-docker.md]:
```suggestion
    - [XRPL Testnet Setup Scripts for Docker](https://github.com/UNIC-IFF/xrpl-docker-testnet)
```

**Review by amarantha-k — APPROVED** [review]
LGTM

**Pull request merged**